### PR TITLE
#159965298 Document api swagger 

### PR DESCRIPTION
**Pull request by pabishai**
**What does this PR do?**
Have documentation of A.P.I endpoints using swagger

**Description of Task to be completed?**
Have documentation of the following endpoints
`POST /api/user/signup`
`POST /api/user/login`

**How should this be manually tested?**
Run `git clone https://github.com/andela/ah-magnificent6.git` from your local machine to clone the repo. 
After cloning the repo, make the repo folder your current working directory. Run `git checkout ft-document-api-swagger-159965298` to go to the relevant branch. 
Create a virtual environment by running `python3 -m venv venv` and `source venv/bin/activate` to activate it. After activating it, run `pip install -r requirements.txt` to install the required packages.
Create a `.env` file from the `.env-sample` and run `python manage.py runserver` to start the app.
Go to the link [http://127.0.0.1:8000/docs](http://127.0.0.1:8000/docs) to view the documentation and test it
The expected payloads are in the format:
User registration: `POST /api/user/signup`
```
{
    "email": "author@user.user",
    "username": "author",
    "password": "goodpassword"
}
```

User Login: `POST /api/user/login`
```
{
    "email": "author@user.user",
    "password": "goodpassword"
}
```

**Any background context you want to provide?**
YAML has been depricated in django-rest-swagger making it necessary to use generics.GenericAPIView and its variations to expose the parameters

**What are the relevant pivotal tracker stories?**
[#159965298](https://www.pivotaltracker.com/story/show/159965298)

**Screenshots**
Default documentation homepage
screen shot 2018-09-03 at 14 53 40

Register User API documentation
screen shot 2018-09-03 at 15 01 10

Login API Documentation
screen shot 2018-09-03 at 15 01 01

### CODE REVIEW

**Comment by meshack-mbuvi** [comment]
Hello @PAbishai , when I follow the steps you have provided to test your feature I encounter this error 

> django.core.exceptions.ImproperlyConfigured: Set the SECRET_KEY environment variable and django.core.exceptions.ImproperlyConfigured: Set the DATABASE_URL environment variable

**Comment by pabishai** [comment]
@meshack-mbuvi It requires you add the secret key to your .env file. I did not include the step. Let me include it.

**Comment by meshack-mbuvi** [comment]
image
On the above screenshot, I have noticed that for email, it shows that the expected data is a string. Is it possible to include the format of the email string so that users know the kind of data to provide? Or we are assuming users know what an email is in this context?

**Comment by micahoriaso** [comment]
@PAbishai  when I run `git clone https://github.com/andela/ah-magnificent6/tree/ft-document-api-swagger-159965298` I get the error message: `fatal: repository 'https://github.com/andela/ah-magnificent6/tree/ft-document-api-swagger-159965298/' not found'
When specifying a git clone URI just use the repo URI.

**Comment by pabishai** [comment]
@micahoriaso and @meshack-mbuvi noted. I have amended the PR based on your input

**Comment by micahoriaso** [comment]
In your how to manually test this section of your PR, you failed to specify the creation and activation of the virtual environment

**Comment by micahoriaso** [comment]
@PAbishai  for each payload kindly specify the endpoint URL

**Comment by pabishai** [comment]
@meshack-mbuvi in the suggestion for the sample text. It's something to look into. Let me see if I can find a solution

**Comment by micahoriaso** [comment]
@PAbishai the documentation works fine, however, the payload structure as provided in the documentation does not reflect the payload as specified in the Pull Request, have you considered other Swagger implementations like [drf-yasg](https://github.com/axnsan12/drf-yasg)?

**Comment by meshack-mbuvi** [comment]
The documentation is fine to me. If you find a solution to my concern raised above, let me know but for now, its fine with me.

**Comment by pabishai** [comment]
@micahoriaso Let me look into the drf-yasg

**Comment by mikenthiwa** [comment]
The documentation works fine so far.
Let's consider using input fields instead of text-field when passing payloads.

**Comment by pabishai** [comment]
@mikenthiwa The nested payload led me to that conclusion. I am looking for a solution

**Comment by Bileonaire** [comment]
@PAbishai 
The documentation is running fine. @mikenthiwa has raised the issue I wanted to raise, I believe you are on it, if you find a solution let us know.

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -29,13 +50,25 @@ def post(self, request):
...
+    """
     permission_classes = (AllowAny,)
     renderer_classes = (UserJSONRenderer,)
     serializer_class = LoginSerializer
 
+
     def post(self, request):
-        user = request.data.get('user', {})
+
+        email = request.data.get('email',None)
```
Comment (parseen254) [inline authors/apps/authentication/views.py]: @PAbishai these two assignments can be made a one-liner i.e `a,b = value[a], value[b]`

Comment (parseen254) [reply authors/apps/authentication/views.py]: same applies for Registration view.

**Review by parseen254 — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/views.py (hunk 1 above)
Comment (pabishai) [reply authors/apps/authentication/views.py]: Got it. Let me amend

Comment (pabishai) [reply authors/apps/authentication/views.py]: @parseendavid Amended.

File: authors/urls.py [hunk 2]
```diff
@@ -16,9 +16,17 @@
...
+# Create schema view.
+# Responsible for generating and rendering the JSON spec and rendering the UI
+schema_view = get_swagger_view(title="Authors Haven API Documentation")
+
 urlpatterns = [
     path('admin/', admin.site.urls),
 
     path('api/', include('authors.apps.authentication.urls', namespace='authentication')),
+
+    path('docs/', schema_view),
```
Comment (parseen254) [inline authors/urls.py]: @PAbishai one last thing let's have the docs at our root path `/` of our API.

Comment (pabishai) [reply authors/urls.py]: On it

Comment (pabishai) [reply authors/urls.py]: @parseendavid Done and pushed

**Review by pabishai — COMMENTED** [review]
@parseendavid The requested changes have been effected.

**Review by parseen254 — APPROVED** [review]

**Pull request merged**